vignettes/introduction.Rmd

```diff
@@ -0,0 +1,335 @@
+---
+title: "Introduction to LaminDB"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Introduction to LaminDB}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+This vignette reproduces the LaminDB [Introduction guide](https://docs.lamin.ai/guide).
+The equivalent **{laminr}** code is included here, for the related text see the associated links.
+This vignette requires the **bionty** Python package to be available.
+
+```{r prepare, include = FALSE}
+# Restore the current instance at the end of the vignette
+current_instance <- laminr::get_current_lamin_instance()
+withr::defer(laminr::lamin_connect(current_instance))
+
+# Restore the current user at the end of the vignette
+current_user <- laminr::get_current_lamin_user()
+withr::defer(laminr::lamin_login(current_user))
+
+laminr::lamin_logout() # Use an anonymous user for this vignette
+```
+
+```{r setup}
+library(laminr)
+```
+
+# Walkthrough
+
+See https://docs.lamin.ai/guide#walkthrough.
+
+```{r walkthrough}
+lamin_init_temp(name = "lamin-intro", modules = c("bionty"))
+```
+
+**NOTE:** We have used a `lamin_init_temp()` to create a temporary instance for this vignette but in most cases you should use regular `lamin_init()`
+
+```{r disconnect, include = FALSE}
+# Disconnect from this instance at the end of the vignette
+withr::defer(laminr::lamin_disconnect())
+```
+
+## Transforms
+
+See https://docs.lamin.ai/guide#transforms.
+
+```{r transforms}
+ln <- import_lamindb()
+ln$track()
+
+ln$Transform$df()
+
+ln$Run$df()
+```
+
+## Artifacts
+
+See https://docs.lamin.ai/guide#artifacts.
+
+```{r artifacts}
+df <- ln$core$datasets$small_dataset1(otype = "DataFrame", with_typo = TRUE)
+df
+
+artifact <- ln$Artifact$from_df(df, key = "my_datasets/rnaseq1.parquet")$save()
+artifact$describe()
+
+artifact$cache()
+
+dataset <- artifact$open()
+as.data.frame(dataset)
+
+artifact$load()
+
+artifact$view_lineage()
+
+df_typo <- df
+levels(df$perturbation) <- c("DMSO", "IFNG")
+df["sample2", "perturbation"] <- "IFNG"
+artifact <- ln$Artifact$from_df(df, key = "my_datasets/rnaseq1.parquet")$save()
+artifact$versions$df()
+```
+
+## Labels
+
+See https://docs.lamin.ai/guide#labels.
+
+```{r labels}
+bt <- import_bionty()
```

Comment:
I think I prefer to adopt `import_module("bionty")` for importing modules, as there are more private custom schemas, it looks more consistent.

Reply:
The specific import functions (`import_lamindb()`, `import_bionty()` etc.) perform extra checks that `import_module()` doesn't do. `import_lamindb()` is especially important because it does some wrapping that is important for some functionality called directly from `ln`. If you use `import_module()` you get the raw Python module object back which will cause some issues. There should probably be some more checks/documentation for this.

I think the specific functions is clearer but if you prefer `import_module()` to be the user interface for everything I can make some changes so that works properly.

Reply:
That'd be awesome if you can make the `import_module()` work, it would be more consistent for users that import their own modules. 🙏